code/modules/clothing/glasses/glasses.dm

```diff
@@ -51,21 +51,21 @@
 /obj/item/clothing/glasses/ui_action_click(mob/user, datum/action/item_action/action)
 	//In case someone in the future adds a non-toggle action to a child type
 	if(istype(action, /datum/action/item_action/toggle))
-		var/datum/action/item_action/toggle/toggle = action
-		toggle.toggled = !activate(user)
-		return
+		activate(user)
+		//Always return TRUE for toggles so that the UI button icon updates
+		return TRUE
 
-	activate(user)
+	return activate(user)
 
 ///Toggle the functions of the glasses
-/obj/item/clothing/glasses/proc/activate(mob/user, silent = FALSE)
-	if(!silent)
+/obj/item/clothing/glasses/proc/activate(mob/user)
+	if(activation_sound && deactivation_sound)
 		playsound(get_turf(src), active ? deactivation_sound : activation_sound, 15)
```

Comment:
Missing one sound will prevent the other from playing.

Reply:
No reason why you'd only have one sound rather than both, but I can change it if you think it's best.

Reply:
There's no inherent reason why you'd have to have to have both sounds. Someone could make glasses that only have an activation sound.